File: college_life/pod/22oct/code.py
def main():
    n,m = list(map(int,input().split()))
    ones = []    
    for i in range(n):
        s = input()
        for j in range(len(s)):
            if(s[j] == '1'):
                ones.append((i,j))
    distances = {}
    for i in range(len(ones)):
        for j in range(i+1,len(ones)):
            x = abs(ones[i][0]-ones[j][0])+abs(ones[i][1]-ones[j][1])
            if x in distances:
                distances[x]+=1
            else:
                distances[x] = 1
    q = int(input())
    for _ in range(q):
        x = int(input())
        if x in distances:
            print(distances[x])
        else:
            print('0')


main()

File: college_life/pod/22oct/code.cpp
#include <bits/stdc++.h>
using namespace std;


int main(){
    int n,m;
    cin>>n>>m;
    vector<pair<int,int>> ones;
    string s;
    for(int i = 0; i < n; i++){
        cin>>s;
        for(int j = 0; j < s.size(); j++){
            if(s[j] == '1'){
                // Finding all the position where there is a 1
                ones.push_back({i,j});
            }
        }
    }        
    unordered_map<int,int> distances;
    for(int i = 0; i < ones.size(); i++){
        for(int j = i+1; j < ones.size(); j++){
            // for every manhattan distance checking if the point exists or not
            distances[abs(ones[i].first-ones[j].first)+abs(ones[i].second-ones[j].second)]++;
        }
    }
    int q;
    cin>>q;
    int x;
    while(q--){
        cin>>x;
        // Number of pairs with given distances
        cout<<distances[x]<<endl;
    }
    return 0;
}